src/transformation/transformers/todo-move-somewhere.ts

```diff
@@ -0,0 +1,59 @@
+import * as ts from "typescript";
+import * as lua from "../../LuaAST";
+import { FunctionVisitor, Visitors } from "../context";
+import { validateAssignment } from "../utils/assignment-validation";
+import { isInDestructingAssignment } from "../utils/typescript";
+
+const transformAssertionExpression: FunctionVisitor<ts.AssertionExpression> = (expression, context) => {
+    if (!ts.isConstTypeReference(expression.type)) {
+        validateAssignment(
+            context,
+            expression,
+            context.checker.getTypeAtLocation(expression.expression),
+            context.checker.getTypeAtLocation(expression.type)
+        );
+    }
+
+    return context.transformExpression(expression.expression);
+};
+
+function isWithinLiteralAssignmentStatement(node: ts.Node): boolean {
+    if (!node.parent) {
+        return false;
+    }
+
+    if (
+        ts.isArrayLiteralExpression(node.parent) ||
+        ts.isArrayBindingPattern(node.parent) ||
+        ts.isObjectLiteralExpression(node.parent)
+    ) {
+        return isWithinLiteralAssignmentStatement(node.parent);
+    } else if (isInDestructingAssignment(node)) {
+        return true;
+    } else if (ts.isBinaryExpression(node.parent) && node.parent.operatorToken.kind === ts.SyntaxKind.EqualsToken) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+// TODO: Consider handling without a visitor since it depends on context too much
+const transformOmittedExpression: FunctionVisitor<ts.OmittedExpression> = node => {
+    const isWithinBindingAssignmentStatement = isWithinLiteralAssignmentStatement(node);
+    return isWithinBindingAssignmentStatement ? lua.createAnonymousIdentifier() : lua.createNilLiteral(node);
+};
+
+export const todoMoveSomewhereVisitors: Visitors = {
```

Comment:
Maybe split this up in statements and expressions and put them in their top-level transformer files?